domain/removal/state/model/offer.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package model
+
+import (
+	"context"
+
+	"github.com/canonical/sqlair"
+
+	"github.com/juju/collections/transform"
+
+	removalerrors "github.com/juju/juju/domain/removal/errors"
+	"github.com/juju/juju/internal/errors"
+)
+
+func (st *State) OfferExists(ctx context.Context, oUUID string) (bool, error) {
```

Comment:
```suggestion
func (st *State) CheckOfferExists(ctx context.Context, oUUID string) error {
```
This is just my opinion, but I find these type of methods to be better when it only returns an error. Absence of an error means the offer exists.
Also, the action (check in this case) should be part of the method name.